azext_edge/edge/vendor/clients/assetmgmt/_client.py

```diff
@@ -0,0 +1,134 @@
+
+from azure.core.pipeline import policies
+from azure.core.rest import HttpRequest, HttpResponse
+from azure.mgmt.core import ARMPipelineClient
+from azure.mgmt.core.policies import ARMAutoResourceProviderRegistrationPolicy
+
+from ._configuration import MicrosoftDeviceRegistryManagementServiceConfiguration
+from ._serialization import Deserializer, Serializer
+from .operations import (
+    AssetEndpointProfilesOperations,
+    AssetsOperations,
+    BillingContainersOperations,
+    OperationStatusOperations,
+    Operations,
+)
+
+if TYPE_CHECKING:
+    from azure.core.credentials import TokenCredential
+
+
+class MicrosoftDeviceRegistryManagementService:
+    """Microsoft.DeviceRegistry Resource Provider management API.
+
+    :ivar operations: Operations operations
+    :vartype operations: deviceregistry.mgmt.operations.Operations
+    :ivar asset_endpoint_profiles: AssetEndpointProfilesOperations operations
+    :vartype asset_endpoint_profiles:
+     deviceregistry.mgmt.operations.AssetEndpointProfilesOperations
+    :ivar assets: AssetsOperations operations
+    :vartype assets: deviceregistry.mgmt.operations.AssetsOperations
+    :ivar billing_containers: BillingContainersOperations operations
+    :vartype billing_containers: deviceregistry.mgmt.operations.BillingContainersOperations
+    :ivar operation_status: OperationStatusOperations operations
+    :vartype operation_status: deviceregistry.mgmt.operations.OperationStatusOperations
+    :param credential: Credential needed for the client to connect to Azure. Required.
+    :type credential: ~azure.core.credentials.TokenCredential
+    :param subscription_id: The ID of the target subscription. The value must be an UUID. Required.
+    :type subscription_id: str
+    :param endpoint: Service URL. Default value is "https://management.azure.com".
+    :type endpoint: str
+    :keyword api_version: Api Version. Default value is "2024-11-01". Note that overriding this
+     default value may result in unsupported behavior.
+    :paramtype api_version: str
+    :keyword int polling_interval: Default waiting time between two polls for LRO operations if no
+     Retry-After header is present.
+    """
+
+    def __init__(
+        self,
+        credential: "TokenCredential",
+        subscription_id: str,
+        endpoint: str = "https://management.azure.com",
+        **kwargs: Any
+    ) -> None:
+        self._config = MicrosoftDeviceRegistryManagementServiceConfiguration(
+            credential=credential, subscription_id=subscription_id, **kwargs
+        )
+        _policies = kwargs.pop("policies", None)
+        if _policies is None:
+            _policies = [
+                policies.RequestIdPolicy(**kwargs),
+                self._config.headers_policy,
+                self._config.user_agent_policy,
+                self._config.proxy_policy,
+                policies.ContentDecodePolicy(**kwargs),
+                ARMAutoResourceProviderRegistrationPolicy(),
```

Comment:
Looks like you are using different version of autorest components to generate this client, compared to our other vendored clients.